frontend/src/screens/wallet/receive/ReceiveOnchain.tsx

```diff
@@ -0,0 +1,425 @@
+          Back to Wallet
+        </LinkButton>
+      </CardFooter>
+    </Card>
+  );
+}
+
+function ReceiveToSpending() {
+  const { toast } = useToast();
+  const { data: info, hasChannelManagement } = useInfo();
+  const { data: balances } = useBalances();
+  const { data: swapFees } = useSwapFees("in");
+  const navigate = useNavigate();
+
+  const [swapAmount, setSwapAmount] = useState("");
+  const [loading, setLoading] = useState(false);
+
+  const onSubmit = async (e: React.FormEvent) => {
+    e.preventDefault();
+
+    try {
+      setLoading(true);
+      const swapInResponse = await request<SwapResponse>("/api/swaps/in", {
+        method: "POST",
+        headers: {
+          "Content-Type": "application/json",
+        },
+        body: JSON.stringify({
+          swapAmount: parseInt(swapAmount),
+        }),
+      });
+      if (!swapInResponse) {
+        throw new Error("Error swapping in");
+      }
+      navigate(`/wallet/swap/in/status/${swapInResponse.swapId}`);
+      toast({ title: "Initiated swap" });
+    } catch (error) {
+      toast({
+        title: "Failed to initiate swap",
+        description: (error as Error).message,
+        variant: "destructive",
+      });
+    } finally {
+      setLoading(false);
+    }
+  };
+
+  if (!info || !balances) {
+    return <Loading />;
+  }
+
+  return (
+    <>
+      <form onSubmit={onSubmit} className="flex flex-col gap-6">
+        {hasChannelManagement &&
+          parseInt(swapAmount || "0") * 1000 >=
+            0.8 * balances.lightning.totalReceivable && (
+            <Alert>
+              <AlertTriangleIcon className="h-4 w-4" />
+              <AlertTitle>Low receiving capacity</AlertTitle>
+              <AlertDescription>
+                You likely won't be able to receive payments until you{" "}
+                <Link className="underline" to="/channels/incoming">
+                  increase your receiving capacity.
+                </Link>
+              </AlertDescription>
+            </Alert>
+          )}
+        <div className="grid gap-1.5">
+          <Label>Amount</Label>
+          <Input
+            type="number"
+            autoFocus
+            placeholder="Amount in satoshis"
+            value={swapAmount}
+            min={MIN_SWAP_AMOUNT}
+            max={(balances.lightning.totalReceivable / 1000) * 0.99}
+            onChange={(e) => setSwapAmount(e.target.value)}
+            required
+            className="[appearance:textfield] [&::-webkit-outer-spin-button]:appearance-none [&::-webkit-inner-spin-button]:appearance-none"
+            endAdornment={
+              <FormattedFiatAmount amount={+swapAmount} className="mr-2" />
+            }
+          />
+          <div className="grid">
+            <div className="flex justify-between text-muted-foreground text-xs sensitive slashed-zero">
+              <div>
+                Receiving Capacity:{" "}
+                {new Intl.NumberFormat().format(
```

Comment:
Make sure you use Math.floor. Please check all places in the project

Reply:
Sorry missed it here, added now!